pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1574,6 +1574,81 @@ func parseU32ConfigItem(newGlobalConfigPtr *types.GlobalConfig, key string,
 
 var itemsPrevConfigHash []byte
 
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+)
+
+type ConfigItemSpec struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intMin     int32
+	intMax     int32
+	intDefault int32
+
+	stringDefault   string
+	boolDefault     bool
+	tristateDefault types.TriState
+}
+
+type ConfigItemSpecMap struct {
+	itemMap map[string]ConfigItemSpec
+}
+
+func (configMap ConfigItemSpecMap) AddIntItem(key string, defaultInt int32, min int32, max int32) {
+	log.Infof("Adding int item %s", key)
+	if defaultInt < min || defaultInt > max {
+		log.Infof("Adding int item %s failed", key)
```

Comment:
Depending on usage (the code which isn't there yet), this should either be an error return or a log.Fatalf.
I suspect the latter, since the set of items are basically specified at compile time.